### PR TITLE
chore: pure pip based better dependencies management

### PR DESCRIPTION
**Pull request by e-cloud**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,13 +2,26 @@
...
+
+Since elastic beanstalk use `requirements.txt` to install packages by default and there is no way to change it.
+We can't use [`constraints.txt` concept](https://stackoverflow.com/questions/34645821/pip-constraints-files) directly for clear dependency organization.
+
+If you want to add new package to this project, follow the steps below:
+
+0. activate the virtual env
+1. `pip install xxx`
+2. add new item for that package in `requirements/base.txt` or `requirements/dev.txt`
+
```
Comment (kunkun-tang) [inline README.md]: Don't we need to adjust requirementx.txt afterward?

File: README.md [hunk 2]
```diff
@@ -2,13 +2,26 @@
...
 ### Steps to run this app
 
 0. Create Environment variables like `QICHACHA_APPKEY` or `TENCENT_APPKEY`(or define them in `.env` file)
 
 ```sh
-pip install --user virtualenv
-pip install -r requirements.txt
+pip install --user virtualenv pre-commit
+pip install -r requirements/base.txt
```
Comment (kunkun-tang) [inline README.md]: should _pip install -r requirements_ be called after env is created?

File: README.md (hunk 1 above)
Comment (e-cloud) [reply README.md]: no, we have a pre-commit hook to update requirements.txt each time.

File: README.md (hunk 2 above)
Comment (e-cloud) [reply README.md]: my bad

**Pull request merged**